### PR TITLE
(HTCONDOR-859) Add workflow to auto Publish/Update/Delete Release News

### PR DESCRIPTION
**Pull request by CannonLock**
Adds workflow that fires when releases are released and adds the release info to htcondor-web. 

Works by using the Github API to get the release info which is dumped into a yaml file at root. This is done with the action posted here -> https://github.com/CHTC/build_release_page/tree/master.

The remaining part of the action ( Line 21+ ) clones the current website and adds the generated yaml file in the appropriate folder to be released on the website.

I have tested using a [forked repo](https://github.com/CannonLock/htcondor) and [test branch on HTCondor-web](https://github.com/htcondor/htcondor-web/commits/preview-test).

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by CannonLock** [comment]
@matyasselmeci I cannot explicitly request a reviewer but can you please take a look at this PR?

File: .github/workflows/publish_release_to_website.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          ssh-private-key: ${{ secrets.HTCONDOR_WEB_DEPLOY_KEY }}
+
+      - run: |
+          mkdir website
+          git clone ${{ env.publish_repo_ssh }} ./website
+          cd website
+          git fetch --all
+          git checkout -b build-release-page origin/${{ env.publish_branch }}
+
+          mv ../*.yml ./${{ env.publish_path }} # Move markdown created by the build_release_page action to branch
```
Comment (matyasselmeci) [inline .github/workflows/publish_release_to_website.yml:33]: Is `../*.yml` correct? It doesn't match the comment.

File: .github/workflows/publish_release_to_website.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Setup SSH Keys and known_hosts
+        uses: webfactory/ssh-agent@v0.5.3
+        with:
+          ssh-private-key: ${{ secrets.HTCONDOR_WEB_DEPLOY_KEY }}
+
+      - run: |
+          mkdir website
+          git clone ${{ env.publish_repo_ssh }} ./website
+          cd website
+          git fetch --all
```
Comment (matyasselmeci) [inline .github/workflows/publish_release_to_website.yml:30]: You don't need this step; you just cloned the repo, it's up to date.

**Review by matyasselmeci — APPROVED** [review]

**Comment by matyasselmeci** [comment]
I left 'regression tests' and 'BaTLab tests' unchecked but they are not applicable to this change.

**Pull request merged**